### PR TITLE
LGA-2448 update end-to-end tests to reflect searching by only postcode

### PR DESCRIPTION
**Pull request by MishNajam**
## What does this pull request do?

FALA no longer allows searching by city. This PR updates our end to end tests to reflect this.

- Most tests that had used "London" have been replaced with a postcode. 
- A test has been added to check that an error is returned when a user searches by city or when an invalid postcode is used
- The search no results test scenario has been edited to check when searching for an organisation that does not exist. This is because when a user searches for any valid postcode, it will always return the nearest available organisation 

## Any other changes that would benefit highlighting?

Added tags on the top of each .feature file to allow us to call specific tests in our repository pipelines

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: behave/features/cla_fala/fala_end_to_end_tests.feature [hunk 1]
```diff
@@ -6,15 +7,25 @@ Feature: FALA end to end tests
...
     Given I provide the "<location>" details
     When I select the "search" button on the FALA homepage
     Then I am taken to the page corresponding to "<location>" result
     Examples:
       | location |
       | SW1H 9AJ |
+
+
+  @fala-search-city @a11y-check
+  Scenario Outline: Search for legal advisers via invalid locations returns error
```
Comment (psweeting1) [inline behave/features/cla_fala/fala_end_to_end_tests.feature:21]: Incorrect postcode is also being tested, so just mention it in the Scenario Outline here.

File: behave/features/steps/fala_end_to_end_tests.py [hunk 2]
```diff
@@ -191,3 +191,16 @@ def step_impl_resulting_generic_search(context, location):
...
+
+
+@step('A "{message}" error is returned')
+def step_impl_postcode_not_found_error_returned(context, message):
+    error_title = context.helperfunc.find_by_css_selector(".govuk-error-summary__title")
+    error_message = context.helperfunc.find_by_css_selector(
+        ".govuk-error-summary__body"
+    )
+
+    assert error_title is not None
```
Comment (psweeting1) [inline behave/features/steps/fala_end_to_end_tests.py:203]: You can chain these asserts together, `assert error_title, error_message is not None`

File: behave/features/steps/fala_end_to_end_tests.py [hunk 3]
```diff
@@ -191,3 +191,16 @@ def step_impl_resulting_generic_search(context, location):
...
+    assert error_message is not None
+    assert error_title.text == "There is a problem"
```
Comment (psweeting1) [inline behave/features/steps/fala_end_to_end_tests.py:205]: Can you add this String to the commons.py, just incase more tests are placed in the future that are checking that String.

File: behave/features/cla_fala/fala_end_to_end_tests.feature [hunk 4]
```diff
@@ -6,15 +7,25 @@ Feature: FALA end to end tests
...
+  Scenario Outline: Search for legal advisers via postcode
     Given I provide the "<location>" details
     When I select the "search" button on the FALA homepage
     Then I am taken to the page corresponding to "<location>" result
     Examples:
       | location |
       | SW1H 9AJ |
+
+
+  @fala-search-city @a11y-check
```
Comment (psweeting1) [inline behave/features/cla_fala/fala_end_to_end_tests.feature:20]: Maybe change the tag to reflect the tests here. Something like `@fala-search-not-valid`

**Review by psweeting1 — COMMENTED** [review]
Looks good, just need to make some small changes then I think it's good to be approved.

**Review by psweeting1 — APPROVED** [review]
LGTM

**Review by Pheonnexx — APPROVED** [review]
LGTM 🚢 🎈

**Pull request merged**